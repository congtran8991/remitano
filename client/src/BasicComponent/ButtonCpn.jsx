import React from "react";

const ButtonCpn = (props) => {
  const { title = "", onClick = () => {}, style = {}, dataTestId = "defaultTestId" } = props;
  return (
    <button
      data-testid={dataTestId}
      style={style}
      onClick={onClick}
      className="btn"
    >
      {title}
    </button>
  );
};

export default ButtonCpn;
