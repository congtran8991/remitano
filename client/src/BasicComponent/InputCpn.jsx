import React from "react";

const InputCpn = (props) => {
  const {
    value = "",
    placeholder = "",
    onChange = () => {},
    name = "",
    type = "text",
    style = {},
    dataTestId = "defaultTestInputId",
  } = props;
  return (
    <input
      data-testid={dataTestId}
      type={type}
      name={name}
      className="input-cpn"
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      style={style}
    />
  );
};

export default InputCpn;
