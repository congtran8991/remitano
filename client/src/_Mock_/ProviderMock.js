import React from "react";
import { Provider } from "react-redux";
import { store } from "../Redux/store";
import { BrowserRouter } from "react-router-dom";

const providerMock = (props) => {
  return (
    <Provider store={store}>
      <BrowserRouter>{props.children}</BrowserRouter>
    </Provider>
  );
};

export default providerMock;
