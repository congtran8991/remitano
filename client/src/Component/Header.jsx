import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import ButtonCpn from "../BasicComponent/ButtonCpn";
import InputCpn from "../BasicComponent/InputCpn";
import { useNavigate } from "react-router-dom";
import {
  userLogin,
  userLoginSlice,
  logoutUser,
} from "../Redux/reducer/userLoginSlice";
import { toast } from "react-toastify";
import { removeCookie, validateEmail } from "../Utils/client";
const Header = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const dataUserLogin = useSelector(userLoginSlice);
  const [infoLogin, setInfoLogin] = useState({ email: "", passWord: "" });
  const handleUserLogin = (e) => {
    const { name, value } = e.target;
    setInfoLogin({ ...infoLogin, [name]: value });
  };
  const handleLoginRegister = () => {
    const { email, passWord } = infoLogin;
    if (!validateEmail(email) || passWord.length === 0) {
      return toast.warn("Email hoặc Password đăng nhập không hợp lệ", {
        position: "top-right",
        autoClose: 3000,
        theme: "light",
      });
    }
    dispatch(userLogin({ email, passWord }));
  };
  return (
    <div className="wrapper-header">
      <div className="header">
        <div className="left-header">Funny Movies</div>
        {!dataUserLogin.isLogin && (
          <div className="right-header">
            <InputCpn
              dataTestId="input-password"
              type="password"
              value={infoLogin.passWord.trim()}
              name="passWord"
              placeholder="Password"
              onChange={handleUserLogin}
            />
            <InputCpn
              dataTestId="input-email"
              type="email"
              value={infoLogin.email}
              name="email"
              placeholder="Email"
              onChange={handleUserLogin}
            />
            <ButtonCpn
              dataTestId="btn-login"
              onClick={handleLoginRegister}
              title="Login/Register"
            />
          </div>
        )}
        {dataUserLogin.isLogin && (
          <div className="right-header">
            <div>Welcome {dataUserLogin?.data?.data?.email}</div>
            <ButtonCpn
              dataTestId="btn-share"
              onClick={() => {
                navigate("/share");
              }}
              title="Share a movie"
            />
            <ButtonCpn
              dataTestId="btn-logout"
              onClick={() => {
                removeCookie("token");
                dispatch(logoutUser());
              }}
              title="Logout"
            />
          </div>
        )}
      </div>
      <hr />
    </div>
  );
};

export default Header;
