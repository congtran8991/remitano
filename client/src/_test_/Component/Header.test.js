/* eslint-disable testing-library/no-unnecessary-act */
import { render, fireEvent, screen, act } from "@testing-library/react";
import Header from "../../Component/Header";
import InputCpn from "../../BasicComponent/InputCpn";
import ButtonCpn from "../../BasicComponent/ButtonCpn";
import ProviderMock from "../../_Mock_/ProviderMock";
// import { act } from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("../../Utils/axios", () => ({
  AXIOS: jest.fn().mockResolvedValueOnce([]),
}));

jest.mock("../../Utils/axios", () => ({
  AXIOS: jest.fn().mockRejectedValueOnce({}),
}));

describe("Header", () => {
  test("Render SharePage", () => {
    const component = render(
      <ProviderMock>
        <Header />
      </ProviderMock>
    ).toJSON;
    expect(component).toMatchSnapshot();
  });
  test("Should update URL video", () => {
    const handleChangeUrl = jest.fn();
    const wrapper = shallow(<InputCpn value="" onChange={handleChangeUrl} />);

    const input = wrapper.find("input");
    const updatedUrl = "https://www.example.com/updated";

    input.simulate("change", { target: { value: updatedUrl } });
    expect(handleChangeUrl).toHaveBeenCalled();
    expect(handleChangeUrl).toHaveBeenCalledWith({
      target: { value: updatedUrl },
    });
  });
  test("Should handle login success", async () => {
    render(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    const inputEmail = screen.getByTestId("input-email");
    const inputPassword = screen.getByTestId("input-password");
    const btnLogin = screen.getByTestId("btn-login");
    await act(async () => {
      fireEvent.change(inputEmail, { target: { value: "congtran@gmail.com" } });
      fireEvent.change(inputPassword, { target: { value: "12345" } });
      fireEvent.click(btnLogin);
    });
  });
  test("Should handle login fail, username or pass not invalid", async () => {
    render(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    const inputEmail = screen.getByTestId("input-email");
    const inputPassword = screen.getByTestId("input-password");
    const btnLogin = screen.getByTestId("btn-login");
    await act(async () => {
      fireEvent.change(inputEmail, { target: { value: "congtran" } });
      fireEvent.change(inputPassword, { target: { value: "" } });
      fireEvent.click(btnLogin);
    });
  });
});
