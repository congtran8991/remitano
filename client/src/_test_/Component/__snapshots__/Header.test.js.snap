// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render SharePage 1`] = `undefined`;
